### PR TITLE
configure user.name, user.email before initial git commit (#179)

### PR DESCRIPTION
**Pull request by marthjod**
Avoid potential failure on initial git commit:
Prompt user for necessary git config values and
provide private key identity names/emails
as suggestions for convenience.

### CODE REVIEW

**Comment by marthjod** [comment]
If this is a way to implement this, I would look into adding tests as well.

**Comment by dominikschulz** [comment]
This looks pretty good, thank you!

If you could add some tests that would be very nice.

File: action/git.go [hunk 1]
```diff
@@ -29,7 +29,19 @@ func (s *Action) gitInit(store, sk string) error {
 		}
 	}
 
-	if err := s.Store.GitInit(store, sk); err != nil {
+	// for convenience, set defaults to user-selected values from available private keys
+	userName, userEmail, err := s.askForGitConfigUser()
```
Comment (marthjod) [inline action/git.go]: **TODO** handle `err`

File: action/clihelper.go [hunk 2]
```diff
@@ -192,6 +192,66 @@ func (s *Action) askForPrivateKey(prompt string) (string, error) {
...
+	keyList = keyList.UseableKeys()
+	if len(keyList) < 1 {
+		return "", "", fmt.Errorf("No usable private keys found")
+	}
+
+	for _, key := range keyList {
+		for _, identity := range key.Identities {
+			useCurrent = false
+
+			if s.isTerm {
```
Comment (dominikschulz) [inline action/clihelper.go]: These nested loops are a little hard to read.
Could you try to unroll them a bit?

Maybe something like
```
if !s.isTerm {
  return identity.Name, identiy.Email
}
useCurrent, err := s.askForBool("Do you want to use name/email for password store x?")
if useCurrent {
  return identity.Name, identiy.Email, nil
}
```
Right now it looks like you could select Name and Email from different identities, which doesn't really make sense.

File: action/clihelper_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	keyList, err := s.gpg.ListPrivateKeys()
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	name, email, _ := s.askForGitConfigUser()
+
+	// tests cannot know whether keyList returned empty or not.
+	// a better distinction would require mocking/patching
+	// calls to s.gpg.ListPrivateKeys()
```
Comment (dominikschulz) [inline action/clihelper_test.go]: For our integration tests we use a `canned` keyring with a fixed set of keys.

File: action/clihelper.go (hunk 2 above)
Comment (marthjod) [inline action/clihelper.go]: I thought this _might_ be an (although not very likely) scenario but it definitely simplifies things to not split identities up, agreed.

File: action/clihelper_test.go (hunk 3 above)
Comment (marthjod) [inline action/clihelper_test.go]: This keyring is used indirectly in [init_test.go#22](https://github.com/justwatchcom/gopass/pull/185/commits/ff0860ed75e819ce1c684070e46969e3c3900817#diff-9e21dcdd5a0a12483a17f3cd3057ef75R22), but I couldn't figure out how to also leverage it in the unit test here.

Comment (dominikschulz) [inline action/clihelper_test.go]: Yes, that's probably too much overhead here. That's why we made part of the tests as integration tests.

File: action/clihelper.go [hunk 4]
```diff
@@ -218,38 +214,22 @@ func (s *Action) askForGitConfigUser() (string, string, error) {
...
-					useCurrent, err = s.askForBool(fmt.Sprintf("Use %q as user name for password store git config?", identity.Name), false)
-					if err != nil {
-						return "", "", err
-					}
-					if useCurrent {
-						name = identity.Name
-					}
+			if !s.isTerm {
+				return identity.Name, identity.Email, nil
+			} else {
```
Comment (dominikschulz) [inline action/clihelper.go]: Almost perfect, but could you please outdent the else branch? Better style.

```
if !s.isTerm {
 return ...
}
useCurrent, ...
```

**Comment by dominikschulz** [comment]
@marthjod If you could rebase real quick, we can merge this. Thanks for your effort!

**Comment by marthjod** [comment]
Done. I could also squash commits for a more concise history if you want.

**Comment by dominikschulz** [comment]
Awesome, thank you very much!

**Pull request merged**